daml-lf/archive/src/stable/protobuf/com/daml/daml_lf_1_15/daml_lf_1.proto

```diff
@@ -0,0 +1,1808 @@
+    StructCon struct_con = 9;
+
+    // Struct project ('ExpStructProj')
+    StructProj struct_proj = 10;
+
+    // Struct update ('ExpStructUpdate')
+    StructUpd struct_upd = 23;
+
+    // Application ('ExpApp')
+    App app = 11;
+
+    // Type Application ('ExpTyApp')
+    TyApp ty_app = 12;
+
+    // Abstraction ('ExpAbs')
+    Abs abs = 13;
+
+    // Type Abstraction ('ExpTyAbs')
+    TyAbs ty_abs = 14;
+
+    // Pattern Matching ('ExpCase')
+    Case case = 15;
+
+    // Let block ('ExpLet')
+    Block let = 16;
+
+    // Empty List ('ExpNil')
+    Nil nil = 17;
+
+    // Non Empty list ('ExpCons')
+    Cons cons = 18;
+
+    // Update expression ('ExpUpdate')
+    Update update = 20;
+
+    // Scenario Expression ('ExpScenario')
+    Scenario scenario = 21;
+
+    // empty optional value ('ExpNone')
+    OptionalNone optional_none = 26;
+
+    // non empty optional value ('ExpSome')
+    OptionalSome optional_some = 27;
+
+    // Wrap an arbitrary value in Any ('ExpToAny')
+    // *Available in versions >= 1.7*
+    ToAny to_any = 30;
+
+    // Extract the given type from Any or return None on type-mismatch ('ExpFromAny')
+    // *Available in versions >= 1.7*
+    FromAny from_any = 31;
+
+    // A type representation
+    // *Available in versions >= 1.7*
+    Type type_rep = 32;
+
+    // Wrap an arbitrary exception into an AnyException ('ExpToAnyException').
+    // *Available in versions >= 1.14*
+    ToAnyException to_any_exception = 33;
+
+    // Extract an arbitrary exception from an AnyException ('ExpFromAnyException').
+    // *Available in versions >= 1.14*
+    FromAnyException from_any_exception = 34;
+
+    // Throw an exception ('ExpThrow').
+    // *Available in versions >= 1.14*
+    Throw throw = 35;
+
+    // Convert to/from interface payloads.
+    // *Available in versions >= 1.15*
+    ToInterface to_interface = 36;
+    FromInterface from_interface = 37;
+
+    // Invoke an interface method.
+    // *Available in versions >= 1.15*
+    CallInterface call_interface = 38;
+
+    // Obtain a contract's signatories/observers through an interface.
+    // *Available in versions >= 1.15*
+    SignatoryInterface signatory_interface = 39;
+    ObserverInterface observer_interface = 40;
+
+    // Invoke an interface method.
+    // *Available in versions >= 1.15*
+    ViewInterface view_interface = 41;
+
+    // Unsafe conversion from interface payloads.
+    // *Available in versions >= 1.1dev*
+    UnsafeFromInterface unsafe_from_interface = 42;
+
+    // Upcast/downcast interface payloads.
+    // *Available in versions >= 1.dev*
+    ToRequiredInterface to_required_interface = 43;
+    FromRequiredInterface from_required_interface = 44;
+    UnsafeFromRequiredInterface unsafe_from_required_interface = 45;
+
+    // Obtain the type representation of a contract's template through an interface.
+    // *Available in versions >= 1.dev*
+    InterfaceTemplateTypeRep interface_template_type_rep = 46;
```

Comment:
```suggestion
    // *Available in versions >= 1.15*
    InterfaceTemplateTypeRep interface_template_type_rep = 43;
```

We'd need to reorder these in `daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf_1.proto`.
I think we could also mark 43-45 as reserved here, and then as the up/downcasting primitives in 1.16.